src/System.Management.Automation/CoreCLR/EventResource.cs

```diff
@@ -0,0 +1,264 @@
+﻿#if UNIX
+/*
+    This code was generated by the tools\ResxGen\ResxGen.ps1 run against PowerShell-Core-Instrumentation.man.
```

Comment:
As long as this is generated, maybe generate a switch statement instead of a dictionary.

Switch dispatch can be very fast (jump table) or a quick binary search, and the other benefit is we won't allocate a bunch of objects that might never be used.

Reply:
Quoted from "https://stackoverflow.com/questions/44905/c-sharp-switch-statement-limitations-why/48259#48259"
> At the end of the day, a C# switch against an integer expression on a modern system is a sub-microsecond operation, and not normally worth worrying about.

Reply:
Fair enough, I'll regenerate the code to use an out parameter for the parameter count and avoid the object allocation.